### PR TITLE
Go: convert regex-use, url-redirection sinks to use models-as-data

### PR DESCRIPTION
**Pull request by owen-mc**
Convert regex-use, url-redirection sinks to use models-as-data. Follows on from https://github.com/github/codeql/pull/17162.

### CODE REVIEW

File: go/ql/lib/semmle/go/frameworks/stdlib/Regexp.qll
Comment (egregius313) [inline go/ql/lib/semmle/go/frameworks/stdlib/Regexp.qll:34]: Does this need to add handling of `-1` corresponding to `receiver` like done in the http concepts library?

**Review by egregius313 — CHANGES_REQUESTED** [review]
Other than my comment about the regex kind, this pretty much LGTM.

However, since this modifies several of the same framework libraries as #17162, I am going to hold off on approving the PR until after the rebasing has happened.

**Comment by owen-mc** [comment]
I hadn't realised that the two PRs conflicted. Now that https://github.com/github/codeql/pull/17162 is merged I have rebased (resolving conflicts) and added a commit addressing your review comment.

**Review by egregius313 — APPROVED** [review]
LGTM provided tests pass

**Pull request merged**
